config/stp.py

```diff
@@ -0,0 +1,951 @@
+
+#
+# 'spanning-tree' group ('config spanning-tree ...')
+#
+
+import click
+import utilities_common.cli as clicommon
+from natsort import natsorted
+import logging
+
+STP_MIN_ROOT_GUARD_TIMEOUT = 5
+STP_MAX_ROOT_GUARD_TIMEOUT = 600
+STP_DEFAULT_ROOT_GUARD_TIMEOUT = 30
+
+STP_MIN_FORWARD_DELAY = 4
+STP_MAX_FORWARD_DELAY = 30
+STP_DEFAULT_FORWARD_DELAY = 15
+
+STP_MIN_HELLO_INTERVAL = 1
+STP_MAX_HELLO_INTERVAL = 10
+STP_DEFAULT_HELLO_INTERVAL = 2
+
+STP_MIN_MAX_AGE = 6
+STP_MAX_MAX_AGE = 40
+STP_DEFAULT_MAX_AGE = 20
+
+STP_MIN_BRIDGE_PRIORITY = 0
+STP_MAX_BRIDGE_PRIORITY = 61440
+STP_DEFAULT_BRIDGE_PRIORITY = 32768
+
+PVST_MAX_INSTANCES = 255
+
+
+def get_intf_list_in_vlan_member_table(config_db):
+    """
+    Get info from REDIS ConfigDB and create interface to vlan mapping
+    """
+    get_int_vlan_configdb_info = config_db.get_table('VLAN_MEMBER')
+    int_list = []
+    for key in get_int_vlan_configdb_info:
+        interface = key[1]
+        if interface not in int_list:
+            int_list.append(interface)
+    return int_list
+
+##################################
+# STP parameter validations
+##################################
+
+
+def is_valid_root_guard_timeout(ctx, root_guard_timeout):
+    if root_guard_timeout not in range(STP_MIN_ROOT_GUARD_TIMEOUT, STP_MAX_ROOT_GUARD_TIMEOUT + 1):
+        ctx.fail("STP root guard timeout must be in range 5-600")
+
+
+def is_valid_forward_delay(ctx, forward_delay):
+    if forward_delay not in range(STP_MIN_FORWARD_DELAY, STP_MAX_FORWARD_DELAY + 1):
+        ctx.fail("STP forward delay value must be in range 4-30")
+
+
+def is_valid_hello_interval(ctx, hello_interval):
+    if hello_interval not in range(STP_MIN_HELLO_INTERVAL, STP_MAX_HELLO_INTERVAL + 1):
+        ctx.fail("STP hello timer must be in range 1-10")
+
+
+def is_valid_max_age(ctx, max_age):
+    if max_age not in range(STP_MIN_MAX_AGE, STP_MAX_MAX_AGE + 1):
+        ctx.fail("STP max age value must be in range 6-40")
+
+
+def is_valid_bridge_priority(ctx, priority):
+    if priority not in range(STP_MIN_BRIDGE_PRIORITY, STP_MAX_BRIDGE_PRIORITY + 1):
+        ctx.fail("STP bridge priority must be in range 0-61440")
+    if priority % 4096 != 0:
```

Comment:
>4096 [](http://example.com/codeflow?start=18&length=4)

Magic number, also, move non multiple of 4096 check to the beginning.

Reply:
Addressed